### ISSUE TITLE
[web-animations-1][web-animations-2] Setting the timeline of an animation

### ISSUE THREAD

**Issue by kevers-google** [w3c/csswg-drafts#5401]
The procedure for setting the timeline of an animation remains in web-animations-1 despite timelines changing to readonly in the spec.  Should this block move to web-animations-2?  Additional steps will be required to support scroll timelines (active project in Blink), but they can wait until after we decide where the rules should reside.

**Comment by flackr** [comment]
Do we know how much work implementing mutable timelines in chromium is going to be? As far as I can tell this is supported in Firefox and Safari per https://wpt.fyi/results/web-animations/timing-model/animations?label=experimental&label=master&aligned and the only reason we made it read-only was because we thought we couldn't get support in chromium.

Perhaps we should be making timeline writable in web-animations-1 and keep the procedure there.

@birtles @graouts does this make sense?

**Comment by kevers-google** [comment]
Blink implementation is in the works and closely aligns with WPT expectations for document timelines and null timelines. The implementation also includes proposed changes specific for non-monotonically increasing timelines. Expect the remaining effort to be minimal.

**Comment by birtles** [comment]
Yes, I think if we are happy with how the behavior for setting timelines works with regards to non-monotonically increasing timelines, it would be fine to restore it to level 1.

**Comment by kevers-google** [comment]
A few tweaks will be needed for non-monotonically increasing timelines, which I intend to put into a followup issue/pull-request. Similar to how play and pause have some specialized logic for setting the start time, setTimeline should align with the scroll position when transitioning to a ScrollTimeline and should likely preserve currentTime if transitioning from a ScrollTimeline to a DocumentTimeline.
